pkg/mock/handlers/handlers.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package handlers
+
+import (
+	"log"
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/aws/amazon-ec2-metadata-mock/pkg/mock/dynamic"
+	"github.com/aws/amazon-ec2-metadata-mock/pkg/mock/static"
+	"github.com/aws/amazon-ec2-metadata-mock/pkg/server"
+)
+
+const (
+	// shortestRouteLength represents smallest metadata path without prefix + "/" ex: "mac/"
+	shortestRouteLength = 4
+	versionsPath        = "/"
+	latestPath          = "/latest"
+)
+
+var (
+	routeLookupTable = make(map[string][]string)
+
+	supportedVersions   = []string{"latest"}
+	supportedCategories = []string{"dynamic", "meta-data"}
+
+	// trimmedRoutes represents the list of routes served by the http server without "latest/meta-data/" prefix
+	trimmedRoutes        []string
+	trimmedRoutesDynamic []string
+)
+
+// CatchAllHandler returns subpath listings, if available; 404 status code otherwise
+func CatchAllHandler(res http.ResponseWriter, req *http.Request) {
+	log.Println("Received request to CatchAllHandler: ", req.URL.Path)
+	// CatchAllHandler may be invoked before ListRoutesHandler
+	if len(trimmedRoutes) == 0 {
+		formatRoutes()
+	}
+
+	var routes, results []string
+
+	// Clean request path and determine which route list to search
+	trimmedRoute := strings.TrimSuffix(req.URL.Path, "/")
```

Comment:
does `trailingSlashMiddleware` handle this?

Reply:
yup you're right, I can initialize to `req.URL.Path` instead